internal/service/advancedclustertpf/resource_compatiblity.go

```diff
@@ -16,6 +16,10 @@ func overrideKnowTPFIssueFields(modelIn, modelOut *TFModel) {
 	if beforeVersion != nil && !modelIn.MongoDBMajorVersion.Equal(modelOut.MongoDBMajorVersion) {
 		modelOut.MongoDBMajorVersion = types.StringPointerValue(beforeVersion)
 	}
+	retainBackups := conversion.NilForUnknown(modelIn.RetainBackupsEnabled, modelIn.RetainBackupsEnabled.ValueBoolPointer())
```

Comment:
While not related to this PR was curious why we can not use the API response value for MongoDBMajorVersion, for reference in SDK v2 we do use the response value but have some additional processing with `FormatMongoDBMajorVersion`.

Reply:
`.mongo_db_version: was cty.StringVal("8"), but now cty.StringVal("8.0")`.
TPF Requires config values to be untouched

Reply:
ohh I see. This is unfortunate as we do lose drift detection for this field, meaning users receives successful apply but not actual guarantee the mongodb major version was changed in the cluster. Might be worth investigating if a plan modifier can help us in this case.

Reply:
Good point, added it to my notes